### PR TITLE
docs: add support for BibTeX references in Sphinx docs

### LINKED ISSUE #4698: References in the documentation
I have seen for other projects, that they provide some references in their documentation. For HPX, I think we should provide some references on the module page. For example, we could provide the references there, how people could cite the module. 

@msimberg Can we use BibTeX in our documentation to do so?

### PR DESCRIPTION
**Pull request by bhumitattarde**
Fixes #4698 

## Proposed Changes

- Adds support for BibTeX references in Sphinx docs
- Added an example citation (this will have to be removed before merging)

## Any other context you'd like to provide?

This is how the references/citations will look with this change:

image

EDIT:
Changed the style of references because earlier style would have meant there would me a single, unified bibliography section. New style should look like this:

image

### CODE REVIEW

File: cmake/templates/conf.py.in [hunk 1]
```diff
@@ -40,6 +40,7 @@ sys.path.insert(0, os.path.abspath('.') + '/extensions')
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
 # ones.
 extensions = [
+    'sphinxcontrib.bibtex',
```
Comment (hkaiser) [inline cmake/templates/conf.py.in:43]: We most likely will have to install this module for our CI environment here: https://github.com/STEllAR-GROUP/docker_build_env/blob/master/hpx_build_env/Dockerfile#L94. Would you mind creating another PR over there as well?

**Comment by bhumitattarde** [comment]
@diehlpk How does this look? 
You can list the references you'd like to provide and their location here. I'll create them.

File: cmake/templates/conf.py.in (hunk 1 above)
Comment (bhumitattarde) [reply cmake/templates/conf.py.in:43]: Thought so, [done](https://github.com/STEllAR-GROUP/docker_build_env/pull/43)!

**Comment by diehlpk** [comment]
@bhumitattarde Looks good. Thanks.

**Comment by diehlpk** [comment]
For APEX, we could add this reference:

K.   A.   Huck,   A.   Porterfield,   N.   Chaimov,H.  Kaiser,  A.  D.  Malony,   T.  Sterling,  andR. Fowler,An  autonomic  performance  environmentfor  exascale,  Supercomputing  frontiers  and  innova-tions, 2 (2015), pp. 49–66

**Comment by diehlpk** [comment]
@hkaiser Can help with different papers about the various modules. I think, if we have a paper about the module or component it would be nice to reference these.

**Comment by bhumitattarde** [comment]
> For APEX, we could add this reference:
> 
> K. A. Huck, A. Porterfield, N. Chaimov,H. Kaiser, A. D. Malony, T. Sterling, andR. Fowler,An autonomic performance environmentfor exascale, Supercomputing frontiers and innova-tions, 2 (2015), pp. 49–66

Done.

> @bhumitattarde I've just merged [STEllAR-GROUP/docker_build_env#43](https://github.com/STEllAR-GROUP/docker_build_env/pull/43). That means you'll now have to update the CI configurations over here to use the new image. Could you find-and-replace all occurrences of `stellargroup/build_env:9` to `stellargroup/build_env:10`?

Done.

**Review by msimberg — APPROVED** [review]
This looks good to me technically. I suppose there are probably more references that could be added, but that's not so easy for @bhumitattarde to judge. More references can always be added as we go.

**Comment by msimberg** [comment]
@hkaiser, @diehlpk do you have any references that you would still like to add at this point or should we go ahead with this as is?

**Comment by diehlpk** [comment]
@msimberg Yes, I will try to add them here by Monday. I am caught with teaching and getting HPX and Octo-Tiger running on Perlmutter.

**Comment by diehlpk** [comment]
@bhumitattarde 

I think the references provided for each component in our JOSS paper could be a good start.

https://joss.theoj.org/papers/10.21105/joss.02352

I think once you added these, we are good for now.

**Review by diehlpk — APPROVED** [review]
LGTME!

**Comment by bhumitattarde** [comment]
> @bhumitattarde
> 
> I think the references provided for each component in our JOSS paper could be a good start.
> 
> https://joss.theoj.org/papers/10.21105/joss.02352
> 
> I think once you added these, we are good for now.

@diehlpk Will look into it and send the PR later.

**Pull request merged**